### ISSUE TITLE
Create smaller Docker images

### ISSUE THREAD

**Issue by stigkj** [rancher/rancher#98]
Would it not be better if you used a smaller FROM image, for examples some sort of busybox-image, like jeanblanchard/busybox-java?

**Comment by anapsix** [comment]
Agreed.. it would be nice to have smaller images.. but it's forgivable while in alpha/beta.. 
though, OpenWRT switched from GLIBC to MUSL, which breaks progrium/busybox :(
I've switched my Java image to be [AlpineLinux based](https://registry.hub.docker.com/u/anapsix/alpine-java), after installing glibc as described here: https://github.com/progrium/busybox/issues/24

**Comment by ibuildthecloud** [comment]
We are contemplating Alpine but it's a good amount of effort to switch.  Any help would be appreciated.

**Comment by anapsix** [comment]
@ibuildthecloud, take a look at the first stab of running rancher-agent over Alpine: [Dockerfile](https://github.com/anapsix/rancher/blob/ea0954a561db95015fe81b78223bab5024853fb5/agent/Dockerfile)

I've had to re-tag my image as `rancher/agent:v0.7.11` before starting it, but it works just fine, seemingly..
Had to install glibc (since Alpine ships with MUSL) to accommodate nsenter hack, but it doesn't bloat the image that much and can be removed once nsenter dependency is no more.

### Usage
1: `docker pull docker pull anapsix/rancher-agent:v0.7.11`
2: `docker tag -f anapsix/rancher-agent:v0.7.11 rancher/agent:v0.7.11`
3. use as you would official agent (get the key from WebUI [Hosts]->[Add Host] section)
or
`cd ./agent; docker build -t rancher/agent:v0.7.11`

**Comment by cloudnautique** [comment]
@anapsix, to make it a little easier to test, when you launch the rancher/server container set:
`-e CATTLE_BOOTSTRAP_REQUIRED_IMAGE=<yourimage>`

That should keep you from having to retag the image.

**Comment by anapsix** [comment]
@cloudnautique sweet!

**Comment by anapsix** [comment]
I'm working on rancher/server now, but having trouble with s6 :(
```
s6-svc: fatal: unable to control /service: supervisor not listening
```

**Comment by cloudnautique** [comment]
Hmm... we are using a pinned version of jprjr/s6-builder, which has since been deprecated. It builds a static s6-binary. Looks like it was replaced by just-containers/skaware... might have to look into giving it a try. I know it was being built statically with Musl in Archlinux. If you want to give it a shot, or I might be able to find some time this week. Keep us posted!

**Comment by anapsix** [comment]
@cloudnautique `-e CATTLE_BOOTSTRAP_REQUIRED_IMAGE=<yourimage>` doesn't work.. 
it's reflected in the UI, but then `rancher/agent-update` kicks in and launches official `rancher/agent` and it all spins indefinitely unable to start..

**Comment by cloudnautique** [comment]
hrm.. It looks like inside your agent image Dockerfile you still have `ENV RANCHER_AGENT_IMAGE rancher/agent:v0.7.11` ? I would change that to your own, that variable is read by run.sh and is likely where the official agent is coming into play.

**Comment by pwFoo** [comment]
Any progress on switching to a smaller base image?

**Comment by composer22** [comment]
+1

**Comment by will-chan** [comment]
Closing this issue.  It will be part of a larger refactoring work on Rancher and Cattle.
